src/PAModel/CanvasDocument.cs

```diff
@@ -309,6 +309,8 @@ internal CanvasDocument(CanvasDocument other)
             _logoFile = other._logoFile != null ? new FileEntry(other._logoFile) : null;
             _entropy = other._entropy.JsonClone();
             _checksum = other._checksum.JsonClone();
+
+            this._idRestorer = new UniqueIdRestorer(this._entropy);
```

Comment:
This looks like a real bug. 
The regular load paths allocate this, was missing from ctor path, and causing null refs.

Reply:
actually, I think this still has an issue, in Merger.ApplyDelta(), we delete entropy via setting to null. That means idRestorer would be pointing to an orphaned entropy object.